beesdoo_shift_swap/models/subscribed_underpopulated_shift.py

```diff
@@ -0,0 +1,282 @@
+        return [
+            ('draft','Draft'),
+            ('validate','Validate'),
+            ('done','Done')
+        ]
+
+    state=fields.Selection(
+        selection=_get_selection_status,
+        default='draft'
+    )
+    worker_id = fields.Many2one(
+        "res.partner",
+        domain=[
+            ("is_worker", "=", True),
+            ("working_mode", "in", ("regular", "irregular")),
+            ("state", "not in", ("unsubscribed", "resigning")),
+        ],
+    )
+    exchanged_timeslot_id = fields.One2many(
+        comodel_name='beesdoo.shift.timeslots_date',
+        inverse_name='id',
+        string='exchanged_shift'
+    )
+    exchange_status = fields.Boolean(default=False, string="status exchange shift")
+    exchanged_shift_id = fields.One2many(
+        comodel_name='beesdoo.shift.shift',
+        inverse_name='id',
+        compute='is_shift_exchanged_already_generated'
+    )
+    comfirmed_timeslot_id = fields.One2many(
+        comodel_name='beesdoo.shift.timeslots_date',
+        inverse_name='id',
+        string='asked_shift'
+    )
+    comfirmed_shift_id = fields.One2many(
+        comodel_name='beesdoo.shift.shift',
+        inverse_name='id',
+        compute='is_shift_comfirmed_already_generated'
+    )
+    comfirme_status = fields.Boolean(default=False, string="status comfirme shift")
+    date = fields.Date(required=True)
+
+    def update_status(self):
+        if self.exchanged_timeslot_id and self.comfirmed_timeslot_id and self.worker_id and self.date:
+            self.write({"state" : "validate"})
+            if self.exchange_status and self.comfirme_status :
+                self.write({"state" : "done"})
+
+    @api.multi
+    def is_shift_exchanged_already_generated(self):
+        # Get current user
+        #cur_user = self.env["res.users"].browse(self.uid)
+
+        #if the supercooperateur make the exchange
+        cur_user = self.worker_id
+
+        # Get current date
+        now = datetime.now()
+
+        # check if the old_shift is already generated
+        # check if the new_shift is already generated
+        if not self.exchanged_shift_id:
+            # Get the shift if it is already generated
+            subscribed_shifts_rec = (
+                self.env["beesdoo.shift.shift"]
+                    .search(
+                    [
+                        ("start_time", ">", now.strftime("%Y-%m-%d %H:%M:%S")),
+                        ('start_time', '=', self.exchanged_timeslot_id.date),
+                        ("worker_id", "=", cur_user.id),
+                        ("task_template_id", "=", self.exchanged_timeslot_id.template_id.id)
+                    ],
+                    limit=1,
+                )
+            )
+
+            # check if is there a shift generated
+            if subscribed_shifts_rec:
+                self.exchanged_shift_id = subscribed_shifts_rec
+                return True
+            return False
+        return True
```

Comment:
Compute function is in fact procedure, they should not return anything. Just set or update the right class attribute. So theses lines can be removed. :)

Reply:
I know, it was just to test the function with a button